Remove some unneeded `Arbitrary` and `Gen` instances

Follow-up to #3772.

ScalaCheck already provides instances for `Duration` and `FiniteDuration`. We should use them and delete ours. If it turns out that the ScalaCheck provided instances don't work for us, we should fix them upstream.

Requires deleting some lines in https://github.com/typelevel/cats/blob/master/kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala.

Removing cats overrides of `Duration` and `FiniteDuration` crash the tests on at least the two following issues:
- integer overflow
```
ScalaTestFailureLocation: org.scalatestplus.scalacheck.CheckerAsserting$$anon$2 at (LawTests.scala:248)
org.scalatest.exceptions.GeneratorDrivenPropertyCheckFailedException: IllegalArgumentException was thrown during property evaluation.
  Message: integer overflow
  Occurred when passed generated values (
    arg0 = 9223372036854775807 nanoseconds,
    arg1 = 9223372036854775807 nanoseconds
  )
	at org.scalatestplus.scalacheck.CheckerAsserting$$anon$2.indicateFailure(CheckerAsserting.scala:251)
	at org.scalatestplus.scalacheck.CheckerAsserting$$anon$2.indicateFailure(CheckerAsserting.scala:238)
	at org.scalatestplus.scalacheck.UnitCheckerAsserting$CheckerAssertingImpl.check(CheckerAsserting.scala:160)
	at org.scalatestplus.scalacheck.Checkers.check(Checkers.scala:371)
	at org.scalatestplus.scalacheck.Checkers.check$(Checkers.scala:369)
	...
Caused by: java.lang.IllegalArgumentException: integer overflow
	at scala.concurrent.duration.FiniteDuration.safeAdd(Duration.scala:614)
	at scala.concurrent.duration.FiniteDuration.add(Duration.scala:619)
	at scala.concurrent.duration.FiniteDuration.$plus(Duration.scala:652)
	at cats.kernel.instances.FiniteDurationGroup.combine(FiniteDurationInstances.scala:42)
```
- `Duration.Undefined` not being equal to itself
```
GeneratorDrivenPropertyCheckFailedException was thrown during property evaluation.
 (LawTests.scala:246)
  Falsified after 22 successful property evaluations.
  Location: (LawTests.scala:246)
  Occurred when passed generated values (
    arg0 = Duration.Undefined
  )
  Label of failing property:
    Expected: Duration.Undefined
Received: Duration.Undefined
```

--- a/kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala
+++ b/kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala
@@ -30,23 +30,6 @@ object KernelCheck {
   implicit val arbitraryUUID: Arbitrary[UUID] =
     Arbitrary(Gen.uuid)
 
-  implicit val arbitraryDuration: Arbitrary[Duration] = {
-    // max range is +/- 292 years, but we give ourselves some extra headroom
-    // to ensure that we can add these things up. they crash on overflow.
-    val n = (292L * 365) / 500
-    Arbitrary(
-      Gen.oneOf(
-        Gen.choose(-n, n).map(Duration(_, DAYS)),
-        Gen.choose(-n * 24L, n * 24L).map(Duration(_, HOURS)),
-        Gen.choose(-n * 1440L, n * 1440L).map(Duration(_, MINUTES)),
-        Gen.choose(-n * 86400L, n * 86400L).map(Duration(_, SECONDS)),
-        Gen.choose(-n * 86400000L, n * 86400000L).map(Duration(_, MILLISECONDS)),
-        Gen.choose(-n * 86400000000L, n * 86400000000L).map(Duration(_, MICROSECONDS)),
-        Gen.choose(-n * 86400000000000L, n * 86400000000000L).map(Duration(_, NANOSECONDS))
-      )
-    )
-  }
-
   implicit val arbitraryFiniteDuration: Arbitrary[FiniteDuration] = {
     // max range is +/- 292 years, but we give ourselves some extra headroom
     // to ensure that we can add these things up. they crash on overflow.
@@ -63,6 +46,10 @@ object KernelCheck {
       )
     )
   }
+
+  // `Duration.Undefined`, `Duration.Inf` and `Duration.MinusInf` break the tests
+  implicit val arbitraryDuration: Arbitrary[Duration] =
+    Arbitrary(arbitraryFiniteDuration.arbitrary.map(fd => fd: Duration))
 
   // Copied from cats-laws.
   implicit def arbitrarySortedMap[K: Arbitrary: Order, V: Arbitrary]: Arbitrary[SortedMap[K, V]] =
@@ -101,17 +88,6 @@ object KernelCheck {
 
   implicit val cogenUUID: Cogen[UUID] =
     Cogen[(Long, Long)].contramap(u => (u.getMostSignificantBits, u.getLeastSignificantBits))
-
-  implicit val cogenDuration: Cogen[Duration] =
-    Cogen[Long].contramap { d =>
-      if (d == Duration.Inf) 3896691548866406746L
-      else if (d == Duration.MinusInf) 1844151880988859955L
-      else if (d == Duration.Undefined) -7917359255778781894L
-      else d.toNanos
-    }
-
-  implicit val cogenFiniteDuration: Cogen[FiniteDuration] =
-    Cogen[Long].contramap(_.toNanos)
 }
 
 class TestsConfig extends ScalaCheckSuite {

